UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TensorApplier.cs

```diff
@@ -30,7 +30,12 @@ public interface IApplier
             /// <param name="agents">
             /// List of Agents that will receive the values of the Tensor.
             /// </param>
-            void Apply(TensorProxy tensorProxy, IEnumerable<Agent> agents);
+            /// <param name="memories">
+            /// The memories of all the agents
+            /// </param>
+            void Apply(TensorProxy tensorProxy,
```

Comment:
Rather than pass this everywhere, can the two classes that actually need it (BarracudaRecurrentInputGenerator and BarracudaMemoryOutputApplier) save a reference to it?